File: backend/packages/common/src/windup_common/enums/media.py
"""媒体文件相关枚举。"""

from enum import StrEnum


class MediaCategory(StrEnum):
    """上传文件的业务分类,用于生成对象存储 key 的目录。

    放在 common 而非 media 模块:新增文件用途时无需修改 media 代码。
    """

    REFERENCE_IMAGE = "reference-image"
    OUTFIT_PREVIEW = "outfit-preview"
    ACTION_FRAME = "action-frame"
    GENERAL = "general"

File: frontend/src/app/adapters/media.ts
import type { MediaReference } from '@/entities'

import { upload as uploadRequest } from './http-client'

/* ─── 后端 DTO ─── */

interface BackendMediaUpload {
  url: string
  object_key: string
  filename: string
  content_type: string
  size: number
}

/* ─── 前端接口 ─── */

export type MediaCategory = 'reference-image' | 'outfit-preview' | 'action-frame' | 'general'

export interface MediaApis {
  upload(file: File, category?: MediaCategory): Promise<MediaReference>
}

/* ─── 适配器 ─── */

export function createMediaApis(): MediaApis {
  return {
    async upload(file: File, category: MediaCategory = 'general'): Promise<MediaReference> {
      const formData = new FormData()
      formData.append('file', file)
      formData.append('category', category)

      const result = await uploadRequest<BackendMediaUpload>('/media/upload', formData)
      return result.url as MediaReference
    },
  }
}
